Code/Legacy/CrySystem/XConsole.cpp

```diff
@@ -2437,100 +2437,49 @@ bool CXConsole::IsOpened()
 }
 
 //////////////////////////////////////////////////////////////////////////
-void CXConsole::PrintLine(const char* s)
+void CXConsole::PrintLine(AZStd::string_view s)
 {
     AddLine(s);
 }
 
 //////////////////////////////////////////////////////////////////////////
-void CXConsole::PrintLinePlus(const char* s)
+void CXConsole::PrintLinePlus(AZStd::string_view s)
 {
     AddLinePlus(s);
 }
 
-static const char* FindNextEndOfLineCharacter(const char* str, size_t length)
-{
-    size_t index = 0;
-    while (index < length)
-    {
-        if ((str[index] == '\r') || (str[index] == '\n'))
-        {
-            return (str + index);
-        }
-
-        index++;
-    }
-
-    return nullptr;
-}
-
 //////////////////////////////////////////////////////////////////////////
-void CXConsole::AddLine(const char* inputStr)
+void CXConsole::AddLine(AZStd::string_view inputStr)
 {
-    if ((inputStr == nullptr) || (inputStr[0] == 0))
+    if (inputStr.empty())
     {
         return;
     }
 
-    size_t totalLen = strlen(inputStr);
-
-    // strip trailing \n or \r.
-    while ((totalLen > 0) && (inputStr[totalLen - 1] == '\n') || (inputStr[totalLen - 1] == '\r'))
-    {
-        totalLen--;
-    }
-
-    // split out each line in a memory efficient way
-    size_t remainingLength = totalLen;
-    const char* lastLine = inputStr;
-    const char* cursor = FindNextEndOfLineCharacter(inputStr, remainingLength);
-    while (cursor != nullptr)
+    // split out each line
+    auto ParseLine = [this](AZStd::string_view line)
     {
-        size_t subStrLength = (cursor - lastLine);
-
-        PostLine(lastLine, subStrLength);
-
-        // bump us up to the cursor + 1 to move past the end of line character
-        remainingLength = remainingLength - (subStrLength + 1);
-        lastLine = (cursor + 1);
-
-        // Find the next non-end of line character
-        while ((remainingLength > 0) && ((*lastLine == '\n') || (*lastLine == '\r')))
-        {
-            remainingLength--;
-            lastLine++;
-        }
-
-        cursor = FindNextEndOfLineCharacter(lastLine, remainingLength);
-    }
-
-    // check for leftover
-    if (remainingLength > 0)
-    {
-        PostLine(lastLine, remainingLength);
-    }
+        PostLine(line.data(), line.size());
+    };
+    AZ::StringFunc::TokenizeVisitor(inputStr, ParseLine, "\r\n");
```

Comment:
question: Should this use a system specific EOL terminator? So it's just `\n` on Linux?

Reply:
This is equivalent to the old logic of splitting lines on \<CR> and \<LF>.
Functional wise, this code should be the same.

Reply:
Okay understood